### ISSUE TITLE
Profiling without Profiling PI [sf#244]

### ISSUE THREAD

**Issue by tomschr** [openSUSE/daps#243]
**Reported by tom_schr on 2014-10-30 08:40 UTC**
Consider the VMDP example in our internal SVN. This document is an article but doesn't use profiling at all.
## Problem

When formatting it with FOP, everything seems to work. However, the images in the document are not shown.

When trying XEP, it gives me the following error messages with the same result:

```
daps -vv -d DC-vmdp pdf --formatter=fop
[...]
[error] Failed to create image file:/local/doc/internal/trunk/vmdp/en/build/.images/color/../../noprofile_remarks/vmdp_qemu-ga.png of type image/png
```
## Current Solution

Tanja and I found the "solution" to this problem: you need to add the PI xml-stylesheet in the main file:

```
<?xml-stylesheet href="urn:x-suse:xslt:profiling:novdoc-profile.xsl" type="text/xml" title="Profiling step"?>
```
## Question

Although the project works know as expected, I think the situation can be improved. I could think of the following, permanent solutions:
1. Check if this PI exists in the main file. If not, abort and give a meaningful error message.
2. Rework the profiling code and fallback to some default behavior. A PI isn't needed.

I guess, the most natural behavior would be (2): the user doesn't have to know that there is a PI needed actually. If this is not possible, try to implement at least (1) to help users and old writers to remember correctly. :)
## Hints
- `daps-xslt/common/get-xml-stylesheet.xsl` enhanced this stylesheet (see [r4351], trunk) to allow to terminate with error exit code != 0. I didn't change the default behaviour, but you can switch to a more restrictive error handling by setting parameter `terminate.on.error` to 1.
- `daps-xslt/contrib/checkconformance.xsl` contains also some checks which can't be validated through our DTD. Could be helpful.

**Comment by tomschr** [comment]
**Commented by tom_schr on 2014-10-30 09:02**
- Description has changed:

Diff:

```

--- old
+++ new
@@ -27,3 +27,9 @@

 I guess, the most natural behavior would be (2): the user doesn't have to know that there is a PI needed actually. If this is not possible, try to implement at least (1) to help users and old writers to remember correctly. :)

+
+Hints
+-----
+
+* `daps-xslt/common/get-xml-stylesheet.xsl` enhanced this stylesheet (see [r4351], trunk) to allow to terminate with error exit code != 0. I didn't change the default behaviour, but you can switch to a more restrictive error handling by setting parameter `terminate.on.error` to 1.
+* `daps-xslt/contrib/checkconformance.xsl` contains also some checks which can't be validated through our DTD. Could be helpful.

```

**Comment by fsundermeyer** [comment]
Fixed with commit @5d89b2ed10f87faafd8a279191b4c068af11546d
